### ISSUE TITLE
Configure proper training parameters for pretraining on MS1M_v3

### ISSUE THREAD

**Issue by vitalwarley** [vitalwarley/research#1]
```python
# https://github.com/deepinsight/insightface/blob/master/recognition/arcface_torch/configs/ms1mv3_r100_lr02.py
from easydict import EasyDict as edict


# make training faster
# our RAM is 256G
# mount -t tmpfs -o size=140G  tmpfs /train_tmp


config = edict()
config.loss = "arcface"
config.network = "r100"
config.resume = False
config.output = None
config.embedding_size = 512
config.sample_rate = 1.0
config.fp16 = True
config.momentum = 0.9
config.weight_decay = 5e-4
config.batch_size = 128
config.lr = 0.2
config.verbose = 2000
config.dali = False


config.rec = "/train_tmp/ms1m-retinaface-t1"
config.num_classes = 93431
config.num_image = 5179510
config.num_epoch = 25
config.warmup_epoch = 0
config.val_targets = ['lfw', 'cfp_fp', "agedb_30"]
```

[training log](https://raw.githubusercontent.com/anxiangsir/insightface_arcface_log/master/ms1mv3_r100_lr02/training.log)

**Comment by vitalwarley** [comment]
Talvez não seja preciso: [código](https://github.com/deepinsight/insightface/blob/e83e8ed4a106df37f585a04922ee140e387ce37e/recognition/arcface_torch/backbones/iresnet.py#L60) e [modelo](https://github.com/deepinsight/insightface/tree/master/recognition/arcface_torch#model-zood.)

**Comment by vitalwarley** [comment]
```
python train.py --data-dir ../datasets/MS1M_v3/ --gpus 1 --max_epochs 1 --model resnet101 --batch-size 48 --task pretrain --lr 0.2 --momentum 0.9 --weight-decay 5e-4 --warmup 0 --cooldown 0 --arcface-s 64 --arcface-m 0.5 --precision 16 --normalize
```

- `--arcface-s` e `--arcface-m` segue o padrão do [código original](https://github.com/deepinsight/insightface/blob/e83e8ed4a106df37f585a04922ee140e387ce37e/recognition/arcface_torch/losses.py#L7).
- Treinamento com FP16 com `--precision 16`.
- _Embeddings_ normalizadas com `--normalize`.

![2022-03-15-085008_406x257_scrot](https://user-images.githubusercontent.com/6365065/158374301-217f04a4-9213-47a0-9d5c-6c6524f9e589.png)

Como o `scheduler` considera o número total de passos de treinamento, essa curva de erro pode mudar quando eu aumentar o número de épocas.

```
Training: 2022-01-19 17:24:38,809-Speed 3491.36 samples/sec   Loss 16.4490   LearningRate 0.1875   Epoch: 0   Global Step: 4000   Fp16 Grad Scale: 131072   Required: 11 hours
Training: 2022-01-19 17:25:21,588-[lfw][4000]XNorm: 22.247099
Training: 2022-01-19 17:25:21,588-[lfw][4000]Accuracy-Flip: 0.99133+-0.00407
Training: 2022-01-19 17:25:21,589-[lfw][4000]Accuracy-Highest: 0.99133
Training: 2022-01-19 17:26:11,180-[cfp_fp][4000]XNorm: 19.285814
Training: 2022-01-19 17:26:11,181-[cfp_fp][4000]Accuracy-Flip: 0.88886+-0.01673
Training: 2022-01-19 17:26:11,182-[cfp_fp][4000]Accuracy-Highest: 0.88886
Training: 2022-01-19 17:26:54,072-[agedb_30][4000]XNorm: 21.806404
Training: 2022-01-19 17:26:54,073-[agedb_30][4000]Accuracy-Flip: 0.93333+-0.01485
Training: 2022-01-19 17:26:54,073-[agedb_30][4000]Accuracy-Highest: 0.93333
```

Essas foram as acurácias originais perto do fim da primeira época. Infelizmente no meu treino inicial a acurácia não saiu de zero, **todavia** eu usei 5126827 amostras para treino (~90% do total) e 51758 (~10%) para validação **sem sobreposição nas classes**.

**Comment by vitalwarley** [comment]
Quase três épocas completas na 2080 Ti 12GB do LAB (RIG1) com os mesmo parametros que usei acima, exceto o batch, que dobrei.

![2022-03-18-235014_2560x1440_scrot](https://user-images.githubusercontent.com/6365065/159104075-9c4d9bdc-700f-4f05-99e9-9f28fbf56e7b.png)

Provavelmente a alta taxa de aprendizado? Na imagem do comentário anterior a perda desce mais rapidamente talvez por causa da taxa de aprendizado que também decai mais rápido dado o número menor de épocas. 

Não é problema do modelo, eu acho... pois eu consigo sobreajustar um batch (TODO: sobreajustar mais de um batch).

![2022-03-19-001106_2518x1342_scrot](https://user-images.githubusercontent.com/6365065/159104523-c841c630-3306-4140-a48a-5ecfa5f76ec9.png)

A perda de validação varia. A perda de treino vai para perto de 0, todavia seu gráfico não foi mostrado por algum motivo obscuro. Refarei mais testes. Uma outra coisa estranha é que a acurácia não saiu de zero... vou investigar se é erro de implementação.

Dúvidas;
 - Devo congelar o backbone?

**Comment by vitalwarley** [comment]
> A perda de validação varia. A perda de treino vai para perto de 0, todavia seu gráfico não foi mostrado por algum motivo obscuro. Refarei mais testes. Uma outra coisa estranha é que a acurácia não saiu de zero... vou investigar se é erro de implementação.

CLI com `--log_every_n_steps=1` resolve o problema de não logar a perda -- o default é 50.

> * Devo congelar o backbone?

Não. A implementação original não o faz.
